client-config/src/main/java/com/palantir/conjure/java/client/config/NodeSelectionStrategy.java

```diff
@@ -27,5 +27,10 @@
     /**
      * For each new request, select the "next" node (in some undefined order).
      */
-    ROUND_ROBIN
+    ROUND_ROBIN,
+
+    /**
+     * Try to pin to a particular node based on NODE_PIN_VALUE header otherwise round robin.
+     */
+    PIN_FROM_REQUEST
```

Comment:
how is this different from PIN_UNTIL_ERROR together with a 308+Location?

Reply:
@ellisjoe @iamdanfox 

The context is that we want to have conjure select which node to run on, but we need to be able to guarantee that a set of requests will be executed on the same node. So we don't want the request to specify the actual node, just ensure it will run on the same node as requests with the same 'NODE_PIN_VALUE' header value.